active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,184 @@
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+### Create a NumberBox
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="simple NumberBox"/>
+```
+
+![](images/numberbox-basic.png)
+
+### Add placeholder text
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="simple NumberBox" Header="Enter equation:" PlaceholderText="A + B * C" />
+```
+
+![](images/numberbox-placeholder-text.png)
+
+### Enable calculation support
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for simple calculations" 
+    Header="Enter equation:" 
+    PlaceholderText="A + B * C" 
+    AcceptsCalculation="True" />
+```
+
+![](images/numberbox-acceptscalculation.png)
+
+### Add increment and decrement stepping
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for items in order" 
+    StepSize="0.1"
+    UpDownButtonsEnabled="True"
+    UpDownDragEnabled="True" />
+```
+
+![](images/numberbox-updownbuttons.png)
+
+### Add mininum and maximum values
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for items in order" 
+    PlaceholderText="1" 
+    MinValue="1"
+    MaxValue="99" />
+```
+
+## Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+## API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+## API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```c++ 
+unsealed runtimeclass NumberBox : Windows.UI.Xaml.Controls.TextBox
+{
+    NumberBox();
+    
+    Double Value;
+    
+    Boolean AcceptsCalculation;
+    Boolean UpDownButtonsEnabled;
+    Boolean HyperDragEnabled;
+    Boolean HyperScrollEnabled;
+    
+    Double StepSize;
+  
+    Boolean AreLeadingZerosTrimmed;
+    Double DecimalPrecision;
+    Boolean DoesInputRound;
+    
+    Double MinValue;
+    Double MaxValue;
+    
+
+    static Windows.UI.Xaml.DependencyProperty ValueProperty{ get; };
```

Comment:
By exposing this ValueProperty, what does this mean for the `Text` property inherited from the `TextBox`? I assume both will still be available and ValueProperty will just be a typed version of the Text.

Reply:
That is correct. Additionally, our intention is to keep this in sync by having them inform each other so that changes to one will result in an automatic update to the other. The purpose of this is to have the String and Double version of the input immediately and conveniently.

Reply:
What will the Value be if Text isn't a number?

Reply:
We'll be leaning on validation to surface error states/messages for non-numerical values and non-legal equations. I'll be speaking with @LucasHaines to align with the broader validation development.

Reply:
Text is the thing you enter and see displayed.

Value is the actually computed value outputted.

The two are kept in sync

Reply:
Exactly, @mdtauk.

Reply:
Text, being a settable property, could be set to a non-number. We should just specify the behavior. Presumably either throw or coerce Value to zero, I lean towards throwing.

Reply:
@MikeHillberg I would suspect validation could be useful with this, when it comes to TextBox

Reply:
Will requesting `Value` forcibly trigger calculation if it hasn't yet been done? I'd hope it would.
If querying `Value` and it's in an error state (of whatever kind-such as invalid calculation entered) I'd expect it to return `Double.NaN` rather than `null` as null can have meaning.

Reply:
@mrlacey, great feedback. I have committed this to the spec here: https://github.com/microsoft/microsoft-ui-xaml-specs/commit/8ff8217b8391516a9d013b45cf469b8ca7b3dcac